src/header.rs

```diff
@@ -0,0 +1,25 @@
+/*
+ * © 2022 Arastoo Bozorgi
+ * All rights reserved.
+ */
+
+use crate::error::Error;
+use byteorder::{BigEndian, ByteOrder};
+
+#[derive(Debug, Default)]
+pub struct Header {
+    pub name: String,
+    pub size: usize,
+    pub start: usize,
+}
+
+impl Header {
+    pub fn header(data: &[u8], start: usize) -> Result<Self, Error> {
```

Comment:
warning: constructor `header` has the same name as the type
  --> src\header.rs:17:5
   |
17 | /     pub fn header(data: &[u8], start: usize) -> Result<Self, Error> {
18 | |         // the first 8 bytes includes the atom size and its name
19 | |         // The size is the entire size of the box, including the size and type header, fields, and all contained boxes.
20 | |         let size = BigEndian::read_u32(&data[0..4]) as usize;
...  |
23 | |         Ok(Self { name, size, start })
24 | |     }
   | |_____^
   |
   = note: `#[warn(clippy::self_named_constructors)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#self_named_constructors